File: 242-valid-anagram/valid-anagram.ts
function isAnagram(s: string, t: string): boolean {
    if (s.length != t.length) return false

    const counterOne = {}
    const counterTwo = {}

    for (let str of s){
        counterOne[str] = (counterOne[str] || 0) + 1
    }

    for (let str of t){
        counterTwo[str] = (counterTwo[str] || 0) + 1
    }

    for (let key in counterOne){
        if(!counterTwo[key] || counterTwo[key] !== counterOne[key]){
            return false
        }
    }
    return true
};

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s_counter = {}
        t_counter = {}

        for c in s:
            s_counter[c] = s_counter.get(c,0) + 1
        for c in t:
            t_counter[c] = t_counter.get(c,0) + 1

        return s_counter == t_counter
        
            